### ISSUE TITLE
[HOLD for payment 2022-05-02] [$500] While trying to open the context menu on mWeb(Safari) on image attachment, it shows a blue box on the app - reported by @parasharrajat

### ISSUE THREAD

**Issue by mvtglobally** [Expensify/App#8246]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Send a message with link on any Chat on Safari mWeb.
2. Hold press the link to open the context Menu.

## Expected Result:
Context Menu should be shown and no blue boxes.

## Actual Result:
No Context menu is shown instead a blue box covers most part of the screen.

## Workaround:
unknown 

## Platform:

Where is this issue occurring?

- Mobile Web

**Version Number:** 1.1.43-0
**Reproducible in staging?:** Y 
**Reproducible in production?:** Y 
**Email or phone of affected tester (no customers):** 
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation  
 Require fixing different component than issue - https://github.com/Expensify/App/issues/8245 
![image - 2022-03-21T001357 307](https://user-images.githubusercontent.com/43995119/159204611-77116ef7-f03b-42f1-865c-083581de1149.png)

**Expensify/Expensify Issue URL:** 
**Issue reported by:** @parasharrajat
**Slack conversation:** https://expensify.slack.com/archives/C01GTK53T8Q/p1646674279978599 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by dylanexpensify** [comment]
Posted! 
Internal: https://www.upwork.com/ab/applicants/1505934572388790272/job-details
External: https://www.upwork.com/jobs/~01a730802d2112b532

**Comment by MelvinBot** [comment]
📣 @rushatgabhane You have been assigned to this job by @melvin-bot[bot]!
Please apply to this job in Upwork and leave a comment on the Github issue letting us know when we can expect a PR to be ready for review 🧑‍💻
Keep in mind: [Code of Conduct](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md) | [Contributing](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md) 📖

**Comment by Beamanator** [comment]
Hmm looks like some of our automation broke when `melvin-bot` came alive 🤔 I'm going to re-add `Help Wanted` and hopefully fix some broken automation today that made `Melvin` think `melvin-bot` actually assigned someone already

**Comment by mollfpr** [comment]
## Proposal

Adding css to display `none` of callout information will be enough to resolve this. There's a function to add `disable-select` to `body`. The class will set some style `-webkit-user-select: none !important` and `-webkit-touch-callout: none !important;` so the callout information from ios will not showing.

Call the function `ControlSelection.block();` before showing the bottom modal and only apply the function on small device and touchscreen device.

https://github.com/Expensify/App/blob/af0f42e2370179885981d6312cde7dd9cbb3ef1f/src/pages/home/report/ReportActionItem.js#L98-L102
```diff
    showPopover(event, selection) {
        // Block menu on the message being Edited
        if (this.props.draftMessage) {
            return;
        }

+        if (this.props.isSmallScreenWidth && canUseTouchScreen()) {
+            ControlSelection.block();
+        }
        ReportActionContextMenu.showContextMenu(
            ContextMenuActions.CONTEXT_MENU_TYPES.REPORT_ACTION,
            event,
            selection,
            this.popoverAnchor,
            this.props.reportID,
            this.props.action,
            this.props.draftMessage,
            this.checkIfContextMenuActive,
            this.checkIfContextMenuActive,
        );
    }
```

### Before

https://user-images.githubusercontent.com/25520267/159535223-aa333b12-7d04-464c-a616-e25e7eda86ba.mov

### After

https://user-images.githubusercontent.com/25520267/159535407-ba81ead5-40ac-456b-8481-83fb16d9d5da.mov

**Comment by mollfpr** [comment]
@parasharrajat the `onPressIn` props are not called when doing the long-press.

**Comment by rushatgabhane** [comment]
> only apply the function on small device and touchscreen device.

the bug will be present on iPad then

**Comment by mollfpr** [comment]
@rushatgabhane Yeah you right!

So I only need to check ‘canUseTouchScreen()’ then.

**Comment by rushatgabhane** [comment]
@mollfpr like Rajat said, it was working before so what exactly broke it?

**Comment by mollfpr** [comment]
@parasharrajat @rushatgabhane So sorry I put the wrong video for the before part 😅

The before video is have same result as the video issue. I’ll send it here ASAP.

**Comment by mollfpr** [comment]
@rushatgabhane It's also happening in iOS Chrome.

I can't what you mean by "it was working before".

**Comment by mollfpr** [comment]
cc @parasharrajat @rushatgabhane 

First, I want to make sure that this issue reproduces when we try to select (long-press) on the user name not when trying to open the context menu on the image attached.

https://user-images.githubusercontent.com/25520267/160226244-f59dce4d-8ebf-40dd-acd7-c26442614d3c.MP4

> Sorry that I wasn't clear enough. If I'm not wrong, chrome on iOS is just a skin for Safari.
I meant why it isn't present on Chrome Android.

It also appears in Firefox iOS. I don't have any good answer why it's not happening in Android, but iOS and Android have different approaches when handling the selection.

https://user-images.githubusercontent.com/25520267/160225758-1c57bb1e-610c-4b82-a115-218f0aac1bfd.MP4

> The assumption is that if you'll checkout to a ~1 month old commit. This issue won't be reproducible.
i.e. this issue wasn't present ~1 month ago, but it is now.

Yes, it's not reproducible before this [PR](https://github.com/Expensify/App/pull/8018), `LongPressGestureHandler` wrapped the other `Pressable` in this case `Pressable` for user name.

**Comment by rushatgabhane** [comment]
> it's not reproducible before this https://github.com/Expensify/App/pull/8018, LongPressGestureHandler wrapped the other Pressable in this case Pressable for user name

Nice find!

Woops, we have a regression from https://github.com/Expensify/App/pull/8018 then.

> It also appears in Firefox iOS

Third party browsers on iOS are using the same rendering engine as safari. This might be the reason.

**Comment by rushatgabhane** [comment]
@mollfpr while your proposal should fix the issue, I'm not sure if it's the best we can do.
I'd like to wait for @parasharrajat's review on this, or I'll ask @ Beamanator to step in.

Thanks for your patience!

**Comment by parasharrajat** [comment]
I am not sure if this is a regression from 8018. I think I tested this before that was merged. But you can confirm this by reverting that PR.

**Comment by rushatgabhane** [comment]
> At last, even if this is. We should try to fix the original implementation instead of patching code.

Totally agree!

Funny thing is I'm not able to repro this on `main`

**Comment by mollfpr** [comment]
> Funny thing is I'm not able to repro this on main - v1.1.46-0, for mWeb Safari.

Try to long-press on the name.

**Comment by rushatgabhane** [comment]
> Try to long-press on the name.

Thanks, that worked. I can repro the issue now. I was clicking on image because the issue title said so.

Anyway, #8018 is not the culprit PR. Because reverting it didn't fix it.

**Comment by mollfpr** [comment]
@rushatgabhane  May I know which version release do you checkout?

**Comment by parasharrajat** [comment]
If long press on the image does not reproduce this bug and as you say clicking on the name does. It feels like issue is not valid. Because long press on name does not open th context menu. 

Main motive of this issue was to fix the  long press action where context menu open up.


So we can close this. Cc. @rushatgabhane

**Comment by rushatgabhane** [comment]
@mollfpr if we checkout to a commit like 67812fab115978d1e44e1b7e757ec96d1e9e0825, this issue isn't present.
So ideally, we wanna know what exact commit casued this issue, and fix it at the root cause.
`git bisect` might be helpful to find the culprit commit. 

https://user-images.githubusercontent.com/29673073/160520918-380f9df9-53e1-4a7c-ae99-6eeba6c03a9c.mov

**Comment by rushatgabhane** [comment]
@parasharrajat actually the context menu is opened on long pressing the name too! And the issue is reproducible. https://github.com/Expensify/App/issues/8246#issuecomment-1079650495

That might be the root cause. Context menu shouldn't open when long pressing the name.

**Comment by dylanexpensify** [comment]
@parasharrajat should we double this price or does it look like we have someone?

**Comment by rushatgabhane** [comment]
@dylanexpensify we don't have a proposal that we wanna move forward with yet.

**Comment by parasharrajat** [comment]
@dylanexpensify Rushat is C+ here. Thanks for asking.

**Comment by dylanexpensify** [comment]
Price doubled!

**Comment by mollfpr** [comment]
@rushatgabhane @parasharrajat 

I checkout the code to this commit https://github.com/Expensify/App/commit/ca3ebd84a944d777fe2379d04b66b373e2bf028e because the component `PressableWithSecondaryInteraction` not yet using `LongPressGestureHandler`, and the issue is not come up.

https://user-images.githubusercontent.com/25520267/160605958-3c8556c2-4519-41f0-b731-e1fbd790380d.mov


And another issue came up on Web, you also can do long-press on `MiniReportActionContextMenu` and the context menu will show up. Also, the issue is producible when doing long-press on the avatar.

https://user-images.githubusercontent.com/25520267/160603059-54ae4a88-0d66-472b-b011-44c80c7f76c7.mov

I can't explain why this is happening, but I think it's because with wrap everything with `LongPressGestureHandler`. 
Maybe this [doc](https://docs.swmansion.com/react-native-gesture-handler/docs/1.10.3/interactions#awaiting-other-handlers) will help.

But if we move this code
https://github.com/Expensify/App/blob/ecb2150026fcf997f84488d096761367c7f67768/src/pages/home/report/ReportActionItem.js#L150-L158

to wrap the `children` variable inside `ReportActionItemSingle` and `ReportActionItemGrouped`
https://github.com/Expensify/App/blob/ecb2150026fcf997f84488d096761367c7f67768/src/pages/home/report/ReportActionItem.js#L174-L176
```js
                                        <ReportActionItemSingle action={this.props.action} showHeader={!this.props.draftMessage}>
                                            <PressableWithSecondaryInteraction
                                                ref={el => this.popoverAnchor = el}
                                                onPressIn={() => this.props.isSmallScreenWidth && canUseTouchScreen() && ControlSelection.block()}
                                                onPressOut={() => ControlSelection.unblock()}
                                                onSecondaryInteraction={this.showPopover}
                                                preventDefaultContentMenu={!this.props.draftMessage}

                                            >
                                                {children}
                                            </PressableWithSecondaryInteraction>
                                        </ReportActionItemSingle>
```

The popover will not be called when doing a long-press on an avatar or the name.

https://user-images.githubusercontent.com/25520267/160605527-e0c36997-fcaa-4a2a-bc7d-5e4e0f24d5b1.mov

**Comment by rushatgabhane** [comment]
@mollfpr perfect! 

🎀 👀 🎀 C+ reviewed
@Beamanator I like @mollfpr's [proposal](https://github.com/Expensify/App/issues/8246#issuecomment-1081781838).

**Comment by rushatgabhane** [comment]
@mollfpr perfect! 

🎀 👀 🎀 C+ reviewed
@Beamanator I like @mollfpr's [proposal](https://github.com/Expensify/App/issues/8246#issuecomment-1081781838).

**Comment by MelvinBot** [comment]
📣 @mollfpr You have been assigned to this job by @Beamanator!
Please apply to this job in Upwork and leave a comment on the Github issue letting us know when we can expect a PR to be ready for review 🧑‍💻
Keep in mind: [Code of Conduct](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md) | [Contributing](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md) 📖

**Comment by mollfpr** [comment]
Thank you, guys! 🎉

I applied to Upwork, and will create the PR ASAP!

**Comment by dylanexpensify** [comment]
How we doing here?

**Comment by mollfpr** [comment]
PR is still on review.

**Comment by mollfpr** [comment]
Raise new PR with a new approach.

cc @rushatgabhane @Beamanator

**Comment by mollfpr** [comment]
@dylanexpensify The Upwork job is expired, can you please create a new one?

**Comment by dylanexpensify** [comment]
@mollfpr of course!

New postings -
internal: https://www.upwork.com/ab/applicants/1521414442738176000/job-details
external: https://www.upwork.com/jobs/~01917d5e367a54c20a

**Comment by dylanexpensify** [comment]
Getting payments sent today!

**Comment by mollfpr** [comment]
@dylanexpensify applied, thanks!

**Comment by dylanexpensify** [comment]
@parasharrajat please apply for reporting bonus 😄

**Comment by parasharrajat** [comment]
Done.

**Comment by dylanexpensify** [comment]
Offers sent!

**Comment by rushatgabhane** [comment]
Danke!

**Comment by dylanexpensify** [comment]
Payments sent, contracts ended, post closed. Thanks everyone!
